### DISCUSSION TITLE
Using a confirm dialog after invoking a rust function

### DISCUSSION THREAD

**Discussion by tataDan** [tauri-apps/tauri#4757] [Q&A] [answered]
In my App.svelte file I have a button as follows:

`<button type="button" on:click={do_order}>Submit</button>`

This button calls this function:

```
function do_order() {
   ....
   invoke("order", {
      name: name,
      phone: phone,
      address: address,
      options: options,
    }).then((response) => (order = response));

    (async () => {
      const confirmed = await confirm(order, "Order Confirmation");    
    })();
}
```
If I click on the Submit button once, then nothing happens.  However, if I click on it a second time, then the confirm dialog appears with the expected order information contained within it.

Any suggestions as how to change the code so that it is only necessary to click on the Submit button once would be appreciated.

**Comment by abisar** [comment, marked as answer]
It looks like you're not waiting for invoke to respond. Try this:
````
async function do_order() {
  ...
  let order = await invoke("order", {
    name, // You don't need to specify key if it's the same as value
    phone: phone,
    address: address,
    options: options,
  });
  let confirmed = await confirm(order, "Order Confirmation");
}
````

**Reply by tataDan** [reply]
The code suggested above (with one minor addition) works.
Without the addition typescript generates the following error: `Argument of type 'unknown' is not assignable to parameter of type 'string'.ts(2345)`.  Apparently we need to inform typescript that order is of type string, not of type unknown.

The code with the needed addition is as follows:

```
async function do_order() {
   ...
  let order: string = await invoke("order", {
      name,
      phone,
      address,
      options,
    });

    let confirmed = await confirm(order, "Order Confirmation");
  }
```
In addition to achieving the desired result, the suggested code is much more readable.

Thank you.
